packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2413,3 +2415,80 @@ const waitForSeq = async (
     };
     deltaManager.on("op", handleOp);
 });
+
+export interface ISummarizerRequestOptions {
+    cache: boolean,
+    reconnect: boolean,
+    summarizingClient: boolean,
+}
+
+/**
+ * Forms a function that will request a Summarizer.
+ * @param loaderRouter - the loader acting as an IFluidRouter
+ * @param lastSequenceNumber - the last sequence number (e.g., from DeltaManager)
+ * @param cache - use cache to retrieve summarizer
+ * @param summarizingClient - is summarizer client
+ * @param reconnect - can reconnect on connection loss
+ */
+export const formRequestSummarizerFn = (
+    loaderRouter: IFluidRouter,
+    options: ISummarizerRequestOptions,
+) => async () => {
+    const { cache, reconnect, summarizingClient } = options;
+    const request: IRequest = {
+        headers: {
+            [LoaderHeader.cache]: cache,
+            [LoaderHeader.clientDetails]: {
+                capabilities: { interactive: false },
+                type: summarizerClientType,
+            },
+            [DriverHeader.summarizingClient]: summarizingClient,
+            [LoaderHeader.reconnect]: reconnect,
+        },
+        url: "/_summarizer",
+    };
+
+    const fluidObject = await requestFluidObject<FluidObject<ISummarizer>>(loaderRouter, request);
+    const summarizer = fluidObject.ISummarizer;
+
+    if (!summarizer) {
+        return Promise.reject(new Error("Fluid object does not implement ISummarizer"));
+    }
+
+    return summarizer;
+};
+
+/**
+ * Requests a Summarizer for the purpose of performing on-demand summary.
+ * @param loader - the loader that resolves the request
+ * @param url - the URL used to resolve the container
+ * @param options - options used to resolve the container
+ * @param reconnect - can reconnect on connection loss
+ * @param summarizingClient - is summarizer client
+ */
+export async function requestOnDemandSummarizer(
```

Comment:
It would be great to be more consistent on naming - requestOnDemandSummarizer & formRequestSummarizerFn are of similar form / usage, yet names differ substantially in usage of verbs (request vs. form, Fn vs. no Fn)

Reply:
The subtle difference is that formRequestSummarizerFn is returning a function, not a summarizer. I think the meaning of the name is, "form a function that requests a summarizer." I think it can now be a private method of SummaryManager. I'd actually like to take "OnDemand" out of "requestOnDemandSummarizer," since the summarizer that it makes is not necessarily limited to the on-demand scenario.